Components/FlightControl/Inc/RocketSM.hpp

```diff
@@ -0,0 +1,295 @@
+    RS_IGNITION,    // Ignition of the ignitors
+    RS_LAUNCH,      // Launch triggered by confirmation of ignition (from ignitor) is nominal : MEV Open Sequence
+
+    //-- BURN --
+    // Vents should stay closed, manual venting NOT ALLOWED
+    // !vents open is definitely not ideal for abort! Best to keep it closed with manual override if we fail here
+    // (can we maybe have the code change the true default state by overwriting EEPROM?)
+    // Ideally we don't want to EVER exceed 7 seconds of burn time (we should store this time at the very least - split into 1 sub-stage for each second if necessary).
+    // For timing we want ~1/10th of a second or better.
+    RS_BURN,        // Main burn (vents closed MEV open) - 5-6 seconds (TBD) :
+                    
+
+    //-- COAST --
+    // Manual venting NOT ALLOWED
+    RS_COAST,       // Coasting (MEV closed, vents closed) - 30 seconds (TBD) ^ Vents closed applies here too, in part. Includes APOGEE
+
+    //-- DESCENT / POSTAPOGEE --
+    // Automatic Venting AND Vent Control ALLOWED
+    RS_DESCENT,  // Vents open (well into the descent)
+    RS_RECOVERY,  // Vents open, MEV closed, transmit all data over radio and accept vent commands
+
+    //-- RECOVERY / TECHNICAL --
+    RS_ABORT,       // Abort sequence, vents open, MEV closed, ignitors off
+    RS_NONE         // Invalid state, must be last
+};
+
+/**
+ * @brief Rocket State Commands, all fall under the GLOBAL_COMMAND -> STATE_ACTION umbrella
+ *
+ *        All commands in this list are only activated if the state allows them.
+ */
+enum RocketStateCommands
+{
+    //-- STATE TRANSITION --
+    RST_ABORT,      // Abort the rocket - transition to ABORT if available
+
+
+    RSC_MAX         // Invalid command, must be last
+};
+
+/**
+ * @brief External Rocket Control Commands, all fall under GLOBAL_COMMAND -> CONTROL_ACTION umbrella
+ *
+ *        State specific commands, must be all in-order to avoid duplicate command IDs
+ */
+enum RocketControlCommands
+{
+    //-- PRE-IGNITION and RECOVERY --
+    RSC_ABORT,       // Transition to ABORT state - available from all states except for IGNITION/LAUNCH/BURN
+    RSC_OPEN_VENT,   // Open the vent valve
+    RSC_CLOSE_VENT,  // Close the vent valve
+    RSC_OPEN_DRAIN,  // Open the drain valve
+    RSC_CLOSE_DRAIN, // Close the drain valve
+    RSC_MEV_CLOSE,   // Forces MEV to close - ONLY supported in states where it is safe to close the MEV
+
+    //-- PRELAUNCH --
+    RSC_BEGIN_FILL, // Transition to the FILL state
+
+    //-- FILL --
+    RSC_ARM_CONFIRM_1,   // Enable first ARM confirmation flag
+    RSC_ARM_CONFIRM_2,   // Enable second ARM confirmation flag
+    RSC_ARM_ACTION,      // Transition to the ARM state
+
+    //-- ARM/IGNITION/LAUNCH/BURN --
+    RSC_MANUAL_OVERRIDE_ENABLE, // For any states with locked out capability. If this flag is set using a command the NEXT command will be allowed to directly control valves/vents
```

Comment:
continued from our discussion last week: I don't think we need to program this logic. The operator is expected to know when they are performing "manual override" actions (with a helpful reminder from the front end)

Reply:
This is one of the things I wanted to clarify. Do we want to allow the MEV to close in launch/burn states by just a single command as-is? I thought it was mentioned (in that rocket state meeting) that it _really_ shouldn't close during the launch/burn states.